### PR TITLE
Fix command line handling

### LINKED ISSUE #7875: Command line handling is wrong sometimes
##### Issue type
 - Bug report
 
##### Minetest version
5.0: e5a37543c

##### Summary
1. `minetest --worldlist wtf-you-type-here` acts just like `minetest --worldlist both`.
2. `--color` treats random strings as `auto`
3. `--world` and `--worldname` don’t force local game if `address` is set in the `mintest.conf` and not overridden by `--address ''`, despite the help saying “implies local game.”

Documentation needs fixing as well, e.g. `--worldlist` is described as implying local game, but actually it implies no game at all =) But that’s another issue.

### PR DESCRIPTION
**Pull request by mmattes**
- --color and --worldlist verify the provided value and error if a wrong value got provided
- command line description for color was differnt on win32 but  code did not handle any differenc- - 
- extended the command line description for world and worldname that it is clear that they only start a local game if used with --go

Fixes #7875

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -135,7 +135,8 @@ int main(int argc, char *argv[])
 		return 0;
 	}
 
-	setup_log_params(cmd_args);
+	if(!setup_log_params(cmd_args))
```
Comment (SmallJoker) [inline src/main.cpp]: Missing space after `if`, see https://dev.minetest.net/Code_style_guidelines

File: src/main.cpp [hunk 2]
```diff
@@ -166,9 +167,15 @@ int main(int argc, char *argv[])
 			list_worlds(true, false);
 		} else if (cmd_args.get("worldlist") == "path") {
 			list_worlds(false, true);
-		} else {
+		} else  if (cmd_args.get("worldlist") == "both") {
```
Comment (SmallJoker) [inline src/main.cpp]: Reduce double space to single space please.

File: src/main.cpp [hunk 3]
```diff
@@ -166,9 +167,15 @@ int main(int argc, char *argv[])
...
 			list_worlds(true, true);
 		}
+		else
```
Comment (SmallJoker) [inline src/main.cpp]: Brackets go to the same line as `else`. See link above.

File: src/main.cpp [hunk 4]
```diff
@@ -258,23 +265,17 @@ static void set_allowed_options(OptionList *allowed_options)
 	allowed_options->insert(std::make_pair("map-dir", ValueSpec(VALUETYPE_STRING,
 			_("Same as --world (deprecated)"))));
 	allowed_options->insert(std::make_pair("world", ValueSpec(VALUETYPE_STRING,
-			_("Set world path (implies local game)"))));
+			_("Set world path (implies local game if used with option --go)"))));
 	allowed_options->insert(std::make_pair("worldname", ValueSpec(VALUETYPE_STRING,
-			_("Set world by name (implies local game)"))));
+			_("Set world by name (implies local game if used with option --go)"))));
```
Comment (SmallJoker) [inline src/main.cpp]: Do you think the part in the parentheses is required? I think it's pretty much self-explaining with the `--go` documentation.

File: src/main.cpp [hunk 5]
```diff
@@ -424,8 +425,10 @@ static void setup_log_params(const Settings &cmd_args)
 			Logger::color_mode = LOG_COLOR_ALWAYS;
 		else if (color_mode == "never")
 			Logger::color_mode = LOG_COLOR_NEVER;
-		else
+		else{
```
Comment (SmallJoker) [inline src/main.cpp]: Space after `else`. Additionally, each block above would need braces due to the Linux kernel style guidelines we use in this project.

File: src/main.cpp (hunk 4 above)
Comment (mmattes) [reply src/main.cpp]: For me it was not self-explaining, i would have hat least expected that the world selection list pre-selects the world i passed to --worldname or --world, but both parameters do not have any effect if they are not used with --go

**Comment by rubenwardy** [comment]
Thanks for your contribution!

**Pull request merged**